src/components/Timeline/utils.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Safely stringify an object for display, handling circular references
+ * and removing __typename fields.
+ */
+export const safeStringify = (data: unknown): string => {
+  if (!data) return "";
+
+  try {
+    const seen = new WeakSet();
+
+    return JSON.stringify(
+      data,
+      (key, value) => {
+        if (key === "__typename") return undefined;
+
+        if (typeof value === "object" && value !== null) {
+          if (seen.has(value)) return "[Circular]";
+
+          seen.add(value);
+        }
+
+        return value;
+      },
+      2,
+    );
+  } catch {
+    return "Unable to serialize";
```

Comment:
This should throw, not return string. This solution introduces nasty bugs exposed to client, that can't be properly caught